src/malloc_hook.c

```diff
@@ -54,7 +54,15 @@ EXTERNAL_API void *reallocarray(void *p, size_t n, size_t s) {
 }
 
 EXTERNAL_API int __posix_memalign(void **r, size_t a, size_t s) {
-    /* All iso_alloc allocations are 8 byte aligned */
+    if(is_pow2(a) == false) {
+        *r = NULL;
+        return EINVAL;
+    }
+
+    if(s < a) {
+        s = a;
```

Comment:
Would this cause `iso_free_size(*r, s)` to fail because we increase the allocation size? I tried this patch's approach normally and it didn't play well with sized allocations, notably C++'s size new/delete

Reply:
It shouldn't, the check in `iso_free_size()` only ensure the size isn't smaller than the size of chunk managed by the zone found for the pointer. Here is a code pointer https://github.com/struct/isoalloc/blob/master/src/iso_alloc.c#L1650